app/Http/Controllers/ProductController.php

```diff
@@ -96,10 +103,17 @@ public function previewEmail(Product $product)
     public function show(Product $product)
     {
         $permissions = [
-            'delete' => Auth::user()->can('delete', $product),
-            'update' => Auth::user()->can('update', $product)
+            'delete' => false,
+            'update' => false
         ];
 
+        if(Auth::user()) {
+            $permissions = [
+                'delete' => Auth::user()->can('delete', $product),
```

Comment:
Very nice. Could also have the following which would prevent the array being assigned twice.
'delete' => Auth::user()?->can('delete', $product)
or
'delete' => Auth::check() && Auth::user()->can('delete', $product)